test/new_relic/agent/transaction/request_attributes_test.rb

```diff
@@ -77,6 +147,113 @@ def test_sets_method_from_request
 
           assert_equal 'POST', attrs.request_method
         end
+
+        def test_by_default_only_a_base_set_of_request_headers_are_captured
+          skip_unless_minitest5_or_above
+
+          with_config(allow_all_headers: false, :'attributes.include' => [], :'attributes.exclude' => []) do
+            attrs = RequestAttributes.new(RACK_REQUEST)
+
+            BASE_HEADERS_MAP.each do |method, definition|
+              assert_equal definition[:value],
+                attrs.send(method),
+                "Expected RequestAttributes##{method} to yield >#{definition[:value]}<, got >#{attrs.send(method)}<"
+            end
+            assert_equal NewRelic::EMPTY_HASH, attrs.other_headers, 'Did not expect to find other headers'
+          end
+        end
+
+        def test_if_allow_all_headers_is_specified_then_allow_them_all
+          skip_unless_rails_gte('5.2')
```

Comment:
Ruby 2.4.10 with Rails 4.2 encountered some failing CI checks, so this check has been added to require either no Rails or Rails >= v5.2.

NOTE: because the failing CI checks were ultimately caused by a less than 100% match with the expectations and not because of any errors encountered, I am currently in favor of not actually restricting `allow_all_headers` itself to Rails v5.2+ and in fact allowing anyone who truly wishes to use agent version 9.x, Rails 4.x, AND the opt-in only `allow_all_headers` to give it a go if they really want to.

Reply:
I think that sounds like a good plan, as long as the limitations for versions that could have issues are documented.

Reply:
Traced the problem to Rack v1. We'll require Rack v2+ for `allow_all_headers` to work properly (but the agent will simply behave as if the option is disabled if enabled with Rack v1, it won't error out or anything). This has been noted in the CHANGELOG.

[7ceb153](https://github.com/newrelic/newrelic-ruby-agent/pull/2155/commits/7ceb15353eeafd13711e5b675e63b70991d5b5d8)